libtiledbsoma/test/test_indexer.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import pandas as pd
+
+from tiledbsoma.IntIndexer import IntIndexer
+
+# tiledbsoma.pytiledbsoma.config_logging("debug")
+
+
+# 1d array to list
+def indexer_test(keys: np.array, lookups: np.array):
+    reindexing_exception: bool = False
+    try:
+        indexer = IntIndexer.map_locations(keys)
+        results = indexer.get_indexer(lookups)
+    except pd.errors.InvalidIndexError:
+        reindexing_exception = True
```

Comment:
Instead of testing the invalid-index and valid cases in the same test, it might be easier to follow if one function tests the good cases, and another tests the bad cases.

Reply:
Since this function now only tests failing cases, can this be shortened to eliminate the extra logic? It seems like it should be able to be something like:

```
def indexer_test_fail(...):
    try:
        index = build_index(keys)
        index.get_indexer(lookups)
        raise AssertionError("should have failed")
    except ...:
        pass

    try:
        pd_index = pd.Index(keys)
        pd_index.get_indexer(lookups)
        raise AssertionError("should have failed")
    except ...:
        pass
```